drivers/clock_control/clock_stm32_ll_common.c

```diff
@@ -662,7 +666,13 @@ static void set_up_fixed_clock_sources(void)
 			}
 		}
 #if STM32_HSI_DIV_ENABLED
+#if defined(CONFIG_SOC_SERIES_STM32L0X)
+		if (STM32_HSI_DIVISOR == 4) {
+			LL_RCC_HSI_EnableDivider();
+		}
```

Comment:
The disable function should also be called in an else. I think it is necessary if for example an MCUBoot enables the divider, but the application wants it disabled

Reply:
I agree, nice catch.